internal/pkg/otel/run.go

```diff
@@ -0,0 +1,138 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package otel
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/converter/expandconverter"
+	"go.opentelemetry.io/collector/otelcol"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/internal/pkg/config"
+	"github.com/elastic/elastic-agent/internal/pkg/release"
+)
+
+const buildDescription = "Elastic opentelemetry-collector distribution"
+
+// IsOtelConfig returns true if file is named:
+//   - otel.(yaml|yml)
+//   - otlp.(yaml|yml)
+//   - otelcol.(yaml|yml)
+//
+// in case of other filename it checks for `agent`, `inputs`, `outputs` keyword. If at least one of these is present
+// it assumes agent config and returns (false, nil).
+// If these keywords are missing it checks for presence of otel specific keywords `exporters`, `receivers`,`processors`
+// or `extensions` in combination with `service` definition. In case these are found (true, nil) is returned.
+// If none of these conditions are satisfied (false, nil) is returned assuming agent config for backward compatibility.
+func IsOtelConfig(ctx context.Context, pathConfigFile string) (bool, error) {
+	fileName := filepath.Base(pathConfigFile)
+	if suffix := filepath.Ext(fileName); suffix != ".yml" && suffix != ".yaml" {
+		return false, nil
+	}
+
+	cleanFileName := strings.TrimSpace(strings.ToLower(strings.TrimSuffix(fileName, filepath.Ext(fileName))))
+	if cleanFileName == "otel" || cleanFileName == "otlp" || cleanFileName == "otelcol" {
+		return true, nil
+	}
+
+	rawConfig, err := config.LoadFile(pathConfigFile)
+	if err != nil {
+		return false, errors.New(err,
+			fmt.Sprintf("could not read configuration file %s", pathConfigFile),
+			errors.TypeFilesystem,
+			errors.M(errors.MetaKeyPath, pathConfigFile))
+	}
+
+	mapConfig, err := rawConfig.ToMapStr()
+	if err != nil {
+		return false, errors.New(err,
+			errors.TypeConfig,
+			errors.M(errors.MetaKeyPath, pathConfigFile))
+	}
+
+	// contains agent definition
+	_, hasAgent := mapConfig["agent"]
+	_, hasInputs := mapConfig["inputs"]
+	_, hasOutputs := mapConfig["outputs"]
```

Comment:
These terms are so generic I worry we may regret this one day. Especially since we aren't in complete control of what keys end up in an otel collector configuration.

I can't see an Elastic Agent configuration ever using the terms `exporters` or `receivers` since we already have names for those, can you simplify into just checking for the presence of those?

Reply:
Although we may need to consider the possibility that we allow otel configuration to co-exist inside of the elastic-agent.yml file one day.

Could we just support configuration detection based on the file name initially?

Reply:
For example this would break if we decided to start using `---` to separate the two configurations in one file.

Reply:
i'm ok with file name